release/pkg/manager/calico/manager.go

```diff
@@ -659,7 +653,11 @@ func (r *CalicoManager) buildReleaseTar(ver string, targetDir string) error {
 	return nil
 }
 
-func (r *CalicoManager) buildContainerImages(ver string) error {
+func (r *CalicoManager) buildContainerImages() error {
+	if !r.buildImages {
```

Comment:
We seem to be inconsistent in this file about whether we check the boolean within the function like this, or if we check it before calling the function.

Personally, I think that calling a function and having it secretly do nothing because of configuration makes the code harder to read, so I'd prefer a model like this:

```go
if r.buildImages {
  r.buildImages()
}
```

We can still have safeguards in the function:

```go
func buildImages() error {
  if !r.buildImages {
    return fmt.Errorf("BUG: Attempted to build images, but not configured to do so")
  }
}
```

Reply:
I put this in here to make it more consistent which is similar to what `publishContainerImages` does

Reply:
OK, we can fix up any other inconsistencies in another PR then.